client/src/views/Feed/SinglePost.vue

```diff
@@ -0,0 +1,305 @@
+              <div class="heart">
+                <i class="far fa-heart fa-lg"
+                :class="{'fas': isLiked}"
+                @click="updateLikes(post)">
+                </i>
+                <p class="likes">{{ this.post.likesCount }}</p>
+              </div>
+              <div class="comments-icon" @click="closePopup(true)">
+                <i class="far fa-comment-alt fa-md"></i>
+                <p class="likes">{{ this.commentsCount(this.post.postId) }}</p>
+              </div>
+            </div>
+            <div class="content">
+              <p
+                ref="caption"
+                :class="[post.caption.length > 40 ? 'caption-in-modal' : 'caption-in-modal']"
+              >
+                <router-link :to="{ path: `/profile/${post.nickname}` }">
+                  <span>{{ post.nickname }}</span>
+                </router-link>
+                {{ post.caption }}
+              </p>
+            </div>
+          </div>
+          </template>
+        </q-card-section>
+      </q-card>
+      </div>
+    </q-dialog>
+  </div>
+</template>
+
+<script>
+import PageComments from '../Comments/PageComments'
+import Avatar from '@/components/Avatar.vue'
+import { updateLikes, incrementViewsCounter } from '../../services/posts'
+import { mapGetters, mapActions } from 'vuex'
+
+export default {
+  name: 'OnePost',
+  props: {
+    post: {
+      type: Object,
+      default: null
+    }
+  },
+  data () {
+    return {
+      isModelVisible: false,
+      dialog: false,
+      likesInfo: {
+        show: false
+      },
+      isLiked: false,
+      showNotAuth: false
+    }
+  },
+  computed: {
+    ...mapGetters({ myProfile: 'profile/myProfile' }),
+    ...mapGetters({ comments: 'comments/myComments' }),
+    ...mapGetters({ commentsCount: 'comments/commentsCount' })
+  },
+
+  methods: {
+    ...mapActions({
+      dislikePost: 'posts/dislikePost',
+      likePost: 'posts/likePost'
+    }),
+    isPostLiked (post) {
+      const likesArr = this.post.likes
+      if (likesArr.includes(localStorage.getItem('profileId'))) {
+        this.isLiked = true
+      } else {
+        this.isLiked = false
+      }
+    },
+    incrementViewsCounter (post) {
+      const postId = post.postId
+      incrementViewsCounter(postId)
+        .then(res => { })
+        .catch(error => {
+          console.log(error)
+        })
+    },
+    updateLikes (post) {
+      const postId = post.postId
+      updateLikes(postId)
+        .then(response => {
+          if (response.data.result === 'dislike') {
+            this.dislikePost(postId)
+            this.isLiked = false
+          } else {
+            this.likePost(postId)
+            this.isLiked = true
+          }
+          return response.data.result
+        })
+        .catch(error => {
+          if (error.response.statusText === 'Unauthorized') {
+            this.showNotif('center')
+            return 'You are not authorized!!!'
```

Comment:
Where do you display 'You are not authorized!!!'?

Reply:
next  function showNotif display this message

Reply:
It is for better understanding,
I could remove it.
?

Reply:
It's unused code, remove it please